pkg/cli/mirror/options.go

```diff
@@ -43,3 +52,41 @@ func (o *MirrorOptions) BindFlags(fs *pflag.FlagSet) {
 		"404/NotFound errors encountered while pulling images explicitly specified in the config "+
 		"will not be skipped")
 }
+
+func (o *MirrorOptions) init() {
+	o.cancelCh = makeCancelCh(syscall.SIGINT, syscall.SIGTERM)
+}
+
+// CancelContext will return a cancellable context and listen for
+// cancellation signals
+func (o *MirrorOptions) CancelContext(parent context.Context) (context.Context, context.CancelFunc) {
+	if o.cancelCh == nil {
+		// return unchanged parent when cancel channel is
+		// not initialized
+		return parent, func() {}
```

Comment:
This seems like a source of future bugs. I would either panic here, or make `init()` atomic with `sync.Once` and call it within `CancelContext()` so the channel is always set up.

Reply:
This is a really good poiint. I can see how not returning a cancellable context from this would cause issues.